manual/manual/tutorials/coreexamples.etex

```diff
@@ -706,7 +706,38 @@ let str : _ format =
 \begin{caml_example}{toplevel}
 Printf.printf str 3 4.5 "string value";;
 \end{caml_example}
-
+"fprintf" function is like "printf" except that it takes an output channel as
+the first argument. Here is how to rewrite the pretty-printer using "fprintf":
+\begin{caml_example}{toplevel}
+let print_expr ppf expr =
+  let open_paren prec op_prec output =
+    if prec > op_prec then Printf.fprintf output "%S" "(" in
```

Comment:
Are you sure you want `%S` instead of `%s` here and below?

Reply:
Actually no, I messed up. Thanks for noticing